tests/integration/models/device-type.spec.ts

```diff
@@ -0,0 +1,121 @@
+// tslint:disable-next-line:import-blacklist
+import * as _ from 'lodash';
+import * as m from 'mochainon';
+import * as parallel from 'mocha.parallel';
+import { balena } from '../setup';
+import { timeSuite } from '../../util';
+import type * as BalenaSdk from '../../..';
+const { expect } = m.chai;
+
+const DEVICE_TYPE_NAME = 'Nvidia D3 TX2';
+const DEVICE_TYPE_SLUG = 'srd3-tx2';
+
+describe('Device Type model', function () {
+	timeSuite(before);
+
+	parallel('balena.models.deviceType.getAll()', function () {
+		it('should get all device types', async function () {
+			const deviceTypes = await balena.models.deviceType.getAll();
+			return expect(deviceTypes).to.not.have.length(0);
+		});
+
+		it('should support a callback', function (done) {
```

Comment:
Let's test callbacks only in one method, since we add support for all methods in the same way.